### PR TITLE
Avoid trying to dis/connect if client is already dis/connected.

### PR DESCRIPTION
**Pull request by juankysoriano**
### Task Requested ###

This is a PR to mitigate the following exception when unbinding the ChromeTabs service. `java.lang.IllegalArgumentException: Service not registered: com.novoda.simplechromecustomtabs.connection.Binder$ServiceConnection@xxxxxx`

This corresponds to an open issue: #16 

### Solution implemented ###

Apparently under some circumstances (not reproducible but noticed by mean of Crash Reporting libs) the application was launching that exception and eventually crashing when trying to unbind from a CustomTabs service. 

In this PR we are hardening the attempt to connect / disconnect from a service by placing an extra check to determine if we have a client connected or not.

**Important Note** I actually ignore if this will fix the issue or not because I have been unable to reproduce it nor by manual testing or by stressing my device with MonkeyRunner. But we can release a beta version and let the affected devs give us some feedback.

### CODE REVIEW

File: library/src/main/java/com/novoda/simplechromecustomtabs/connection/Connection.java [hunk 1]
```diff
@@ -14,4 +14,6 @@
 
     void disconnectFrom(@NonNull Activity activity);
 
+    boolean isDisconnected();
```
Comment (ataulm) [inline library/src/main/java/com/novoda/simplechromecustomtabs/connection/Connection.java]: is it too much to add this to the interface? Maybe not when we can Java 8 and have default impl, but could be considered unnecessarily verbose til then

Comment (juankysoriano) [inline library/src/main/java/com/novoda/simplechromecustomtabs/connection/Connection.java]: Yeah I was hesitating about it but decided to include it to have certain symmetry. `connectTo` , `disconnectTo`,  `isConnected`,  isDisconnected`...

**Comment by rock3r** [comment]
LGTM @juankysoriano 

As for the crash we get, it look like the packagemanager thought our service was somehow disabled? Do we ever do that? Or might it be related to Chrome/GMS being updated in the background? That would make such service temporarily unavailable.

**Comment by juankysoriano** [comment]
@rock3r not related. I think the original issue is that, for unknown conditions, the context given for unbinding is different to the context given for binding. 

I have hardened the lib to avoid crashes and used application context for those two operations.

**Comment by blundell** [comment]
:+1:

**Comment by filipmaelbrancke** [comment]
Just wanted to give you ( @juankysoriano  & @blundell & @rock3r )  some feedback:
we could not reproduce the error on any of our test devices, but we had quite a lot of bug reports through Fabric.
After reviewing your changes for v0.1.3-rc1 we've decided to deploy it in one of our apps.
We have now a staged rollout running that is already deployed on close to 20.000 devices, and we see no simple-chrome-custom-tabs related crashes anymore.
So, thanks for the changes!

**Comment by juankysoriano** [comment]
Thank you @filipmaelbrancke 👍  , I will probably promote that from just simply `rc` to a concrete version on the following days.

**Pull request merged**